src/App/DataSource/DataSourceManager.php

```diff
@@ -38,6 +39,13 @@ class DataSourceManager
         'twitter' => Twitter\Twitter::class,
     ];
 
+    /**
+     * The registered custom data sources.
+     *
+     * @var array
+     */
+    protected $customSources = [];
```

Comment:
Just asking about naming here... with "custom" do we mean datasources that are installed in as additional packages?

Reply:
Yes ... data sources installed as a package

Reply:
I would be inclined to call them "add-on" or "plug-in" datasources. To signify they are off the repo tree. Anyway, let's chat later, I'm really curious about the dynamic nature of the code below.